x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -368,8 +381,52 @@ public Map<String, Object> domainUsageStats() {
         }
     }
 
-    public Map<RealmConfig.RealmIdentifier, Authentication.RealmRef> getRealmRefs() {
-        return realmRefs;
+    /**
+     * Retrieves the {@link Authentication.RealmRef}, which contains the {@link DomainConfig}, if configured,
+     * for the passed in {@link RealmConfig.RealmIdentifier}.
+     * If the realm is not currently configured, {@code null} is returned.
+     */
+    public @Nullable Authentication.RealmRef getRealmRef(RealmConfig.RealmIdentifier realmIdentifier) {
+        // "file", "native", and "reserved" realms may be renamed, but they refer to the same corpus of users
+        if (FileRealmSettings.TYPE.equals(realmIdentifier.getType())) {
+            return getFileRealmRef();
+        } else if (NativeRealmSettings.TYPE.equals(realmIdentifier.getType())) {
+            return getNativeRealmRef();
```

Comment:
The kink here is that the name of the native or file realms is irrelevant, when considering the domain configuration, see also https://github.com/elastic/elasticsearch/blob/c1d0e8ed2a8d977ee70faddce69d571c79c693c8/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java#L664 . Moreover there is always exactly one RealmRef for both file and native realms (which may contain the domain conf). The rationale is that the file and native realms always refer to the same corpus of internal users.